### PR TITLE
Updates and reorganization of deployment guides

### PR DESCRIPTION
**Pull request by brooke-hamilton**
# Description

This PR contains documentation updates, including:
- Consolidation of Bicep deployment instructions into a new `docs/deployment-guide-bicep.md` file.
- Moved and updated the terraform deployment instructions into `docs/deployment-guide-terraform.md`.
- Implemented a standard table of contents in the Terraform and Bicep deployment docs.
- Updated the `README.md` docs at `src/terraform` and `src/bicep` to link to the deployment guides.
- Generated a list of parameters for the Bicep folder documentation at `src/bicep/README.md`.
- Removed unused images at `docs/images/custom_template_deployment.gif`, `docs/images/missionlz_as_of_july2021.jpg`, and `docs/images/missionlz_as_of_may2021.png`

## Issue reference

The issue this PR will close: #562
The issue this PR will close: #428
The issue this PR will close: #299 (See the Azure Firewall Premium section in `deployment-guide-bicep.md`.)

## Checklist

Please make sure you've completed the relevant tasks for this PR out of the following list:

* [x] All acceptance criteria in the backlog item are met
* [x] The documentation is updated to cover any new or changed features
* [x] Manual tests have passed
* [x] Relevant issues are linked to this PR

### CODE REVIEW

File: docs/deployment-guide-terraform.md [hunk 1]
```diff
@@ -0,0 +1,424 @@
+# Mission LZ Terraform
```
Comment (glennmusa) [inline docs/deployment-guide-terraform.md:1]: for consistency 
```suggestion
# Mission LZ Deployment Guide for Terraform
```

File: docs/deployment-guide-bicep.md [hunk 2]
```diff
@@ -0,0 +1,433 @@
...
+Azure Security Center offers a standard/defender sku which enables a greater depth of awareness including more recomendations and threat analytics. You can enable this higher depth level of security in MLZ by setting the parameter `deployASC` during deployment. In addition you can include the `emailSecurityContact` parameter to set a contact email for alerts.
+
+Parameter name | Default Value | Description
+-------------- | ------------- | -----------
+`deployASC` | 'false' | When set to "true", enables Azure Security Center for the subscriptions used in the deployment. It defaults to "false".
+`emailSecurityContact` | '' | Email address of the contact, in the form of john@doe.com
+
+#### Azure Sentinel
+
+Azure Sentinel can be enabled by setting the `deploySentinel=true` parameter.
```
Comment (glennmusa) [inline docs/deployment-guide-bicep.md:103]: Sentinel looks like the only optional feature that doesn't have a description of its benefits

```suggestion
[Sentinel](https://docs.microsoft.com/en-us/azure/sentinel/overview) is a scalable, cloud-native, security information and event management (SIEM) and security orchestration, automation, and response (SOAR) solution. Sentinel can be enabled by setting the `deploySentinel=true` parameter.
```

File: docs/deployment-guide-bicep.md [hunk 3]
```diff
@@ -0,0 +1,433 @@
...
+
+Parameter name | Default Value | Description
+-------------- | ------------- | -----------
+`deploySentinel` | 'false' | When set to "true", enables Microsoft Sentinel within the Log Analytics Workspace created in this deployment. It defaults to "false".
+
+#### Remote access with a Bastion Host
+
+If you want to remotely access the network and the resources you've deployed you can use [Azure Bastion](https://docs.microsoft.com/en-us/azure/bastion/) to remotely access virtual machines within the network without exposing them via Public IP Addresses.
+
+To deploy a virtual machine as a jumpbox into the network without a Public IP Address using Azure Bastion Host, provide these parameters:
```
Comment (glennmusa) [inline docs/deployment-guide-bicep.md:113]: Right now, end users get both virtual machines:

```suggestion
Deploy a Linux and Windows virtual machine as jumpboxes into the network without a Public IP Address using Azure Bastion Host by providing values for these parameters:
```

File: docs/deployment-guide-terraform.md [hunk 4]
```diff
@@ -0,0 +1,424 @@
...
+-------------- | ------------- | -----------
+`hub_vnet_address_space` | '10.0.100.0/24' | The address space to be used for the virtual network
+`hub_client_address_space` | '10.0.100.0/26' | The address space to be used for the Firewall virtual network
+`hub_management_address_space` | '10.0.100.64/26' | The address space to be used for the Firewall virtual network subnet used for management traffic
+`tier0_vnet_address_space` | '10.0.110.0/26' | VNet prefix for tier 0
+`tier0_subnets`.address_prefixes | '10.0.110.0/27' | Subnet prefix for tier 0
+`tier1_vnet_address_space` | '10.0.115.0/26' | VNet prefix for tier 1
+`tier1_subnets`.address_prefixes | '10.0.115.0/27' | Subnet prefix for tier 1
+`tier2_vnet_address_space` | '10.0.120.0/26' | VNet prefix for tier 2
+`tier2_subnets`.address_prefixes | '10.0.120.0/27' | Subnet prefix for tier 2
```
Comment (glennmusa) [inline docs/deployment-guide-terraform.md:70]: I don't know if Terraform supports assigning a value to a single property like `address_prefies` on a map of object like `tier0_subnets` -- does it?

https://github.com/Azure/missionlz/blob/9099c3e83e930620ec6ebe8381b4a7e666dddd39/src/terraform/mlz/variables.tf#L338-L340

Can someone specify `-var="tier0_subnets.address_prefixes=foo"`?

If this is possible, great, we'd just need to fix these cells in the table:

```suggestion
`tier0_subnets.address_prefixes` | '10.0.110.0/27' | Subnet prefix for tier 0
`tier1_vnet_address_space` | '10.0.115.0/26' | VNet prefix for tier 1
`tier1_subnets.address_prefixes` | '10.0.115.0/27' | Subnet prefix for tier 1
`tier2_vnet_address_space` | '10.0.120.0/26' | VNet prefix for tier 2
`tier2_subnets.address_prefixes` | '10.0.120.0/27' | Subnet prefix for tier 2
```

If it's not possible, would we document how to modify the parameter like we do below for `create_policy_assignment`?

File: docs/deployment-guide-terraform.md [hunk 5]
```diff
@@ -0,0 +1,424 @@
...
+      source  = "hashicorp/time"
+      version = "0.7.2"
+    }
+  }
+}
+```
+
+To find more information about setting the backend see [Local Backend](https://www.terraform.io/docs/language/settings/backends/local.html),  if you wish to AzureRM backend please see [AzureRM Backend](https://www.terraform.io/docs/language/settings/backends/azurerm.html)
+
+## Cleanup
```
Comment (glennmusa) [inline docs/deployment-guide-terraform.md:382]: Should we document `destroy`ing in the reverse order of `apply`?

e.g. 

1. apply mlz/main.tf
2. apply tier3/main.tf
3. destroy tier3/main.tf
4. destroy mlz/main.tf

**Review by glennmusa — CHANGES_REQUESTED** [review]

File: docs/deployment-guide-terraform.md (hunk 4 above)
Comment (brooke-hamilton) [reply docs/deployment-guide-terraform.md]: I added instructions above the table saying to edit the `variables.tf` files to override the defaults.

File: docs/deployment-guide-terraform.md (hunk 5 above)
Comment (brooke-hamilton) [reply docs/deployment-guide-terraform.md:382]: I added a statement and an example for destroying in reverse order of creating. 🥇

**Review by glennmusa — APPROVED** [review]
🎉

**Pull request merged**